Automattic/jetpack
# Boost: Add health checks for cache loading

Fixes #36331

## Proposed changes:
* Run cache setup on every dashboard page load to self heal any issues
* Add a to site health if page cache is active but not loading
* Show a notice in boost settings, if page cache is active but not loading.

### Other information:


## Jetpack product discussion
None

## Does this pull request change what data or activity we track or use?
None

## Testing instructions:
* Enable boost cache
* Mutilate the caching system, i.e. Set `WP_CACHE` to `false` or delete the `advanced-cache.php` file.
* After loading boost settings, it should auto-fix the issues
* Now, try modifying the advanced-cache.php file slightly so that the cache cannot load. Example: Add a `return;` immidiately after the `<?php` line.
* Load boost dashboard.

<img width="823" alt="image" src="https://github.com/Automattic/jetpack/assets/3737780/234fd95d-58aa-44f7-b89c-783404fd60ca">
